qemu/tests/watchdog.py

```diff
@@ -282,6 +286,26 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def watchdog_test_suit():
+        """
+        Run watchdog-test-framework to verify the function of emulated watchdog
+        devices.
+        """
+
+        _watchdog_device_check(test, session, watchdog_device_type)
+        test_lib = params["test_lib"]
+        test_lib_path = os.path.join(data_dir.get_deps_dir(), test_lib)
+        vm.copy_files_to(test_lib_path, "/home")
+        try:
+            session.cmd_output("/home/%s --yes" % os.path.basename(test_lib_path),
+                               timeout=120)
+        except ShellTimeoutError:
+            o = session.get_output().splitlines()[-1]
+            if 27 <= int(o.rstrip("...")) <= 32:
```

Comment:
Would you please help explain what this line implements?

Reply:
Here are test steps and results of the framework. Watchdog action is pause for the case.
So after the watchdog is triggered, the guest will pause. I just want to get the pause time to judge whether the watchdog is triggered.   
/home/watchdog-test --yes

Welcome to the watchdog test framework.
You should read the README file and run this in the guest.
DO NOT RUN IT IN THE HOST!

The test is as follows:
(1) I will set up the watchdog with a 30 second timeout.
(2) I will ping the watchdog for 60 seconds.  During this time
    the guest should run normally.
(3) I will stop pinging the watchdog and just count up.  If the
    virtual watchdog device is set correctly, then the watchdog
    action (eg. reboot) should happen around the 30 second mark.

Setting up the watchdog (/dev/watchdog) with a 30 second timeout.
Timeout is set to 30 seconds.

Pinging the watchdog for 60 seconds ...

0...
ping
3...
ping
6...
ping
9...
ping
12...
ping
15...
ping
18...
ping
21...
ping
24...
ping
27...
ping
30...
ping
33...
ping
36...
ping
39...
ping
42...
ping
45...
ping
48...
ping
51...
ping
54...
ping
57...
ping
60...
ping


Stopping pings.
The watchdog action should happen at approximately 30 second mark.

0...
3...
6...
9...
12...
15...
18...
21...
24...
27...

Reply:
@YanhuiMa Thanks for the explanation, I don't figure out a better check point so far, but I still think that it is worth to keep this test here. so I would accept your current implementation for now. But I would request you adding comments to the code to explain the test. Thanks.